data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -669,6 +756,246 @@ class ApiServerClient {
+      });
+      resourceBody.status = statusJson;
+    }
+    // Create Namespace if not default
+    const namespaceId = this.getNamespaceId(opts.resourceType === CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEBINDINGS ?
+      _.get(opts, 'spec.instance_id') : opts.resourceId
+    );
+    // Create Namespace if not default
+    return Promise.try(() => this.init())
+      .then(() => this.createNamespace(namespaceId))
+      .then(() => apiserver
+        .apis[opts.resourceGroup][CONST.APISERVER.API_VERSION]
+        .namespaces(namespaceId)[opts.resourceType].post({
+          body: resourceBody
+        }))
+      .catch(err => {
+        return convertToHttpErrorAndThrow(err);
+      });
+  }
+
+
+
+  /**
+   * @description Update OSB Resource in Apiserver with the opts
+   * @param {string} opts.resourceGroup - Name of resource group ex. backup.servicefabrik.io
+   * @param {string} opts.resourceType - Type of resource ex. defaultbackup
+   * @param {string} opts.resourceId - Unique id of resource ex. backup_guid
+   * @param {string} opts.metadata - Metadata of resource
+   * @param {Object} opts.spec - Value to set for spec field of resource
+   * @param {string} opts.status - status of the resource
+   */
+  updateOSBResource(opts) {
+    logger.silly('Updating resource with opts: ', opts);
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to update resource`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to update resource`);
+    assert.ok(opts.resourceId, `Property 'resourceId' is required to update resource`);
+    assert.ok(opts.metadata || opts.spec || opts.status, `Property 'metadata' or 'options' or 'status'  is required to update resource`);
+    return Promise.try(() => {
+        const patchBody = {};
+        if (opts.metadata) {
+          patchBody.metadata = opts.metadata;
+        }
+        if (opts.spec) {
+          patchBody.spec = camelcaseKeys(opts.spec);
+        }
+        if (opts.status) {
+          const statusJson = {};
+          _.forEach(opts.status, (val, key) => {
+            if (key === 'state') {
+              patchBody.metadata = _.merge(patchBody.metadata, {
+                labels: {
+                  'state': val
+                }
+              });
+            }
+            statusJson[key] = _.isObject(val) ? JSON.stringify(val) : val;
+          });
+          patchBody.status = statusJson;
+        }
+        logger.info(`Updating - Resource ${opts.resourceId} with body - ${JSON.stringify(patchBody)}`);
+        // Create Namespace if not default
+        const namespaceId = opts.namespaceId ? opts.namespaceId : this.getNamespaceId(opts.resourceType === CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEBINDINGS ?
+          _.get(opts, 'spec.instance_id') : opts.resourceId
+        );
+        // Create Namespace if not default
+        return Promise.try(() => this.init())
+          .then(() => this.createNamespace(namespaceId))
```

Comment:
why is namespace being created during udpate operation? SHouldnt there be one already and throw exception if not present?

Reply:
I agree. @amitmalav can you please check?

Reply:
Removed